client.go

```diff
@@ -1524,12 +1576,235 @@ func (cl *Client) AddTorrentInfoHash(infoHash metainfo.Hash) (t *Torrent, new bo
 	}
 	new = true
 	t = cl.newTorrent(infoHash)
+	t.updateWantPeersEvent()
+	return
+}
+
+type TorrentState struct {
+	Version int `json:"version"`
+
+	// metainfo or these
+	InfoHash *metainfo.Hash `json:"hash,omitempty"`
+	Name     string         `json:"name,omitempty"`
+	Trackers [][]string     `json:"trackers,omitempty"`
+
+	MetaInfo *metainfo.MetaInfo `json:"metainfo,omitempty"`
+	Pieces   []bool             `json:"pieces,omitempty"`
+
+	Checks []bool `json:"checks,omitempty"`
+
+	// Stats bytes
+	Downloaded int64 `json:"downloaded,omitempty"`
+	Uploaded   int64 `json:"uploaded,omitempty"`
+
+	// dates
+	AddedDate     int64 `json:"added_date,omitempty"`
+	CompletedDate int64 `json:"completed_date,omitempty"`
+
+	// time
+	DownloadingTime int64 `json:"downloading_time,omitempty"`
+	SeedingTime     int64 `json:"seeding_time,omitempty"`
+
+	// .torrent
+	Comment   string `json:"comment,omitempty"`
+	Creator   string `json:"creator,omitempty"`
+	CreatedOn int64  `json:"created_on,omitempty"`
+}
+
```

Comment:
All this state stuff seems specific to your application, and can be done external to this package.